src/gui/toolbarMenubar/model/ToolbarItem.cpp

```diff
@@ -27,6 +27,8 @@ ToolbarItem::~ToolbarItem() = default;
 
 auto ToolbarItem::getName() -> string { return this->name; }
 
+void ToolbarItem::setName(const string& name) { this->name = name; }
```

Comment:
```suggestion
void ToolbarItem::setName(string name) { this->name = std::move(name); }
```

Reply:
For the caller both approaches would yield the same behaviour, right?

```cpp
ToolbarItem toolbarItem{};
std::string name{"test"};
toolbarItem.setName(name);
REQUIRE(name == "test");
```

* `ToolbarItem::setName(const string& name)` passes name by reference but then copy assigns `this->name = name;`
* `ToolbarItem::setName(string name)` passes name by value (hence I guess copying it) and then we move it into `this->name = std::move(name);`

It that's the case I would prefer the `const string& name` as it clearly states that the string name will not be changed in the process. But I guess there's something else I'm not aware of :sweat_smile:

Reply:
> It that's the case I would prefer the `const string& name` as it clearly states that the string name will not be changed in the process. But I guess there's something else I'm not aware of sweat_smile

Both variants guarantee, that the original value is not changed in the process.
But "pass by value and move" has the advantage, that this allows the user to prevent any copies at all, since it moves the point of copy directly in front of the function. Also, it marks the function as a value copying function.

This allows the following:

```
    ToolbarItem toolbarItem{};
    std::string name{"test"};
    toolbarItem.setName(std::move(name));   //< No copy here! name is invalidated
    return toolbarItem;
} // delete empty / invalidated name 
```
But removing the `std::move`:

```
    ToolbarItem toolbarItem{};
    std::string name{"test"};
    toolbarItem.setName(name);   //< Implicit Copy here! name will still be unchanged
    return toolbarItem;
} // destroy name
```

Reply:
Now I get it

1. Approach one: "pass by value"
     a. "... and copy": one copy, hence less efficient; `name` still intact be used after calling setName
     b. "... and move": no copy more,  hence efficient; `name` no longer useful as empty
2. Approach two: "pass by reference"
     a. one copy, hence less efficient; `name` still intact be used after calling setName

So basically by using "pass by value" in the function signature you give the programmer the option whether to move or copy depending on the usecase, whereas with "pass by reference" you force the copy on them :bulb:

Reply:
Will be done :heavy_check_mark: